lib/_stream_wrap.js

```diff
@@ -47,12 +44,12 @@ function StreamWrap(stream) {
     self.emit('error', err);
   });
   this.stream.on('data', function ondata(chunk) {
-    if (!(chunk instanceof Buffer)) {
+    if (typeof chunk === 'string' || this._readableState.objectMode === true) {
```

Comment:
Streams implementors often change `objectMode` after the fact, so we cannot assume that. It'd a semver-major change, and several of us are -1 on those.